### DISCUSSION TITLE
Compilation error

### DISCUSSION THREAD

**Discussion by highfreq** [HefnySco/OBAL#41] [Trouble Shooting]
During compilation of GY-91 patched rover i get the following error:

In file included from ../../libraries/GCS_MAVLink/GCS_MAVLink.h:41,
                 from ../../libraries/AP_AccelCal/AP_AccelCal.h:14,
                 from ../../libraries/AP_InertialSensor/AP_InertialSensor.h:16,
                 from ../../libraries/AP_AHRS/AP_AHRS_Backend.h:26,
                 from ../../libraries/AP_AHRS/AP_AHRS_DCM.h:24,
                 from ../../libraries/AP_AHRS/AP_AHRS.h:32,
                 from ../../libraries/AP_NavEKF2/LogStructure.h:4,
                 from ../../libraries/AP_Logger/LogStructure.h:126,
                 from ../../libraries/AP_Camera/AP_Camera.h:10,
                 from ../../libraries/AP_Camera/AP_Camera_Backend.h:25,
                 from ../../libraries/AP_Camera/AP_Camera_Logging.cpp:1:
libraries/GCS_MAVLink/include/mavlink/v2.0/all/../ardupilotmega/../common/../minimal/./mavlink_msg_heartbeat.h: In function ‘uint16_t mavlink_msg_heartbeat_pack(uint8_t, uint8_t, mavlink_message_t*, uint8_t, uint8_t, uint8_t, uint32_t, uint8_t)’:
libraries/GCS_MAVLink/include/mavlink/v2.0/mavlink_types.h:158:56: internal compiler error: Segmentation fault
 #define _MAV_PAYLOAD_NON_CONST(msg) ((char *)(&((msg)->payload64[0])))
                                                        ^~~~~~~~~
libraries/GCS_MAVLink/include/mavlink/v2.0/all/../ardupilotmega/../common/../minimal/./mavlink_msg_heartbeat.h:88:16: note: in expansion of macro ‘_MAV_PAYLOAD_NON_CONST’
         memcpy(_MAV_PAYLOAD_NON_CONST(msg), &packet, MAVLINK_MSG_ID_HEARTBEAT_LEN);
                ^~~~~~~~~~~~~~~~~~~~~~
0x76b3f10f ???
	../sysdeps/unix/sysv/linux/arm/sigrestorer.S:64
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <file:///usr/share/doc/gcc-8/README.Bugs> for instructions.

I run it

./waf configure --board obal
./waf ardurover

If i run a couple of times the compilation succedes.

**Comment by HefnySco** [comment]
sorry for this late reply. Please let me know if you still have this problem with latest versions.
